### DISCUSSION TITLE
Make `schema.prisma` column names abstract

### DISCUSSION THREAD

**Discussion by pbelbin** [prisma/prisma#2766] [Q&A] [answered]
I see some discussion about the `schema.prisma`, and column naming standards.

Mostly they appear to be focused on making the column names conform to a certain case standard (eg: camelCase, PascalCase, snake_case or kebab-case), so that the generated client code will conform to the preferred case standard for that language.

But, what if you need to generate multiple client languages for the same schema?

Yeah, ok, I suppose you could post-process the 'base' schema into a language-specific schema, and do the per-language generation from that, but that seems like it shouldn't really need to be the case.

I propose the following:

1. provide a setting in the schema that indicates what it's column and entity name standard is meant to be, and the introspection tool and other tooling would then be able to always know what to generate (including relevant `@map` and `@@map`) and expect to be maintained across multiple introspection cycles.

2. let the per-language client generators also have a configuration option that dictates how the client generator will emit column and entity names.

This will allow a project manager to choose how they want to see the `schema.prisma`, and allow the per-language preferences to be distinct, so that, for example, a TypeScript client can use camelCase, and a PHP client can use snake_case, from the same `schema.prisma` file.

There *might* also be a case for having per-database variant (eg: MySQL vs PostgreSQL) mapping variants, but it seems that you're mostly targeting SQL compliant databases, so I'm kind of expecting the need for this to be way less than the per-language variation in naming standards.

I suspect there is a need also to do similar per-language case preferences for the model/entity names too, for the generated client  eg: a project might want TypeScript entity names to be PascalCased, but column names to be camelCase.

Sometimes database table names might be pluralized while the entity name should be singular, too.

For total flexibility, you could even provide some kind of 'translation function definition' that would allow you to custom-define how you want the translations to go, so that special cases can be accommodated for, which the tools would call to actually do the translations.

**Comment by ryands17** [comment, marked as answer]
Hey @pbelbin :wave: 
I think this best be converted to a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) referencing this discussion so that we can look into this further :)

**Comment by ludralph** [comment]
Hi @pbelbin,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
